galaxy_ng/app/api/v3/serializers/namespace.py

```diff
@@ -0,0 +1,118 @@
+import logging
+import re
+
+from django.db import transaction
+
+from rest_framework.exceptions import ValidationError
+from rest_framework import serializers
+
+from galaxy_ng.app import models
+from galaxy_ng.app.models import auth as auth_models
+from galaxy_ng.app.api.v3.serializers.group import GroupSummarySerializer
+
+log = logging.getLogger(__name__)
+
+
+class NamespaceLinkSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.NamespaceLink
+        fields = ('name', 'url')
+
+
+class NamespaceSerializer(serializers.ModelSerializer):
+    links = NamespaceLinkSerializer(many=True, required=False)
+
+    groups = GroupSummarySerializer(many=True)
+
+    class Meta:
+        model = models.Namespace
+        fields = (
+            'id',
+            'name',
+            'company',
+            'email',
+            'avatar_url',
+            'description',
+            'links',
+            'groups',
+            'resources'
+        )
+
+    # replace with a NamespaceNameSerializer and validate_name() ?
+    def validate_name(self, name):
+        if not name:
+            raise ValidationError(detail={
+                'name': "Attribute 'name' is required"})
+        if not re.match(r'^[a-zA-Z0-9_]+$', name):
+            raise ValidationError(detail={
+                'name': 'Name can only contain [A-Za-z0-9_]'})
+        if len(name) <= 2:
+            raise ValidationError(detail={
+                'name': 'Name must be longer than 2 characters'})
+        if name.startswith('_'):
+            raise ValidationError(detail={
+                'name': "Name cannot begin with '_'"})
+        return name
+
+    def create(self, validated_data):
+        log.debug('serializer.create validated_data=%s', validated_data)
+
+        links_data = validated_data.pop('links', [])
+        groups_data = validated_data.pop('groups', None)
+
+        instance = models.Namespace.objects.create(**validated_data)
+
+        ngs = auth_models.Group.objects.in_bulk([x['name'] for x in groups_data], field_name="name")
```

Comment:
Doesn't seem you use `ngs` as dictionary anywhere. So why are you using `in_bulk()` method?
Naming: What does `ngs` mean?

```suggestion
        ngs = auth_models.Group.objects.filter(name__in=[x['name'] for x in groups_data])
```